frontend/packages/noobaa-storage-plugin/src/components/status-card/statuses.tsx

```diff
@@ -1,77 +1,45 @@
-import * as _ from 'lodash';
+import { PrometheusHealthHandler } from '@console/plugin-sdk';
 import { HealthState } from '@console/shared/src/components/dashboard/status-card/states';
-import { PrometheusResponse } from '@console/internal/components/graphs';
-import { FirehoseResult } from '@console/internal/components/utils';
 import { getGaugeValue } from '../../utils';
 
-const NooBaaStatus = [
-  {
-    state: HealthState.ERROR,
-    message: 'MCG is not running',
-  },
-  {
+const nooBaaStatus = {
+  '0': { state: HealthState.OK },
+  '1': {
     state: HealthState.ERROR,
     message: 'All resources are unhealthy',
   },
-  {
+  '2': {
     state: HealthState.WARNING,
     message: 'Object Bucket has an issue',
   },
-  {
+  '3': {
     state: HealthState.ERROR,
     message: 'Many buckets have issues',
   },
-  {
+  '4': {
     state: HealthState.WARNING,
     message: 'Some buckets have issues',
   },
-];
+};
 
-export const getNooBaaState: GetObjectServiceStatus = (
-  prometheusResponses = [],
-  hasLoadError,
-  isLoading,
-  k8sResponse,
-) => {
-  const [buckets, unhealthyBuckets, pools, unhealthyPools] = prometheusResponses.map((r) =>
-    getGaugeValue(r),
-  );
-  const noobaaPhase = _.get(k8sResponse, 'data[0].status.phase');
-  const unhealthyBucketsRatio = unhealthyBuckets / buckets;
-  const noData = !(buckets && unhealthyBuckets && pools && unhealthyPools && noobaaPhase);
+export const getNooBaaState: PrometheusHealthHandler = (responses, noobaa) => {
+  const { response, error } = responses[0];
+  const noobaaLoaded = noobaa?.loaded;
+  const noobaaLoadError = noobaa?.loadError;
+  const statusIndex: string = getGaugeValue(response);
 
-  if (hasLoadError) {
+  if (error || noobaaLoadError) {
     return { state: HealthState.UNKNOWN };
   }
-  if (noData) {
-    return { state: HealthState.NOT_AVAILABLE };
-  }
-  if (isLoading) {
+  if (!noobaaLoaded || !response) {
     return { state: HealthState.LOADING };
   }
-  if (noobaaPhase !== 'Ready') {
-    return NooBaaStatus[0];
-  }
-  if (Number(pools) === Number(unhealthyPools)) {
-    return NooBaaStatus[1];
-  }
-  if (Number(unhealthyBuckets) === 1) {
-    return NooBaaStatus[2];
-  }
-  if (unhealthyBucketsRatio >= 0.5) {
-    return NooBaaStatus[3];
-  }
-  if (unhealthyBucketsRatio >= 0.3) {
-    return NooBaaStatus[4];
+  if (!statusIndex) {
+    return { state: HealthState.NOT_AVAILABLE };
   }
```

Comment:
Is it still required? Why can't we say `unknown`?

I can't assume this happening.

Reply:
In case of prometheus cache, it will hit. The query is present but the gauge value is absent.
Its for those cases when we dont have gauge value `""` and we can say its not available.
We are doing the same for data resiliency/details card. I noticed this before hence put it.

Also not including it with the all errors check above since initially when cluster is being installed then `statusIndex` will be `undefined`. the condition gets true and `Not available` is shown , due to which loading states checks are skipped , hence keeping it after loading check.